src/components/FeatureFlagRenderer.js

```diff
@@ -23,14 +23,22 @@ export default class FeatureFlagRenderer extends Component {
   }
 
   componentDidMount () {
-    this._checkFeatureFlag();
+    this.checkFeatureFlag();
+    this.listenFlagChangeEvent();
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return (
```

Comment:
Does this fix any unwanted behavior, or is it just an extra check?

Reply:
As I was debugging and verifying the change events being emitted, I had thrown some console statements at each lifecycle. I noticed that the feature flag components were getting re-rendered multiple times when they didn't need to. Figured I might as well optimize it since its fairly straight forward to do for this component.